apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -352,6 +399,38 @@ export const collectionRouter = router({
         })
       },
     ),
+
+  getCollectionTags: protectedProcedure
+    .input(getCollectionTagsSchema)
+    .query(async ({ ctx, input: { resourceId, siteId } }) => {
+      await bulkValidateUserPermissionsForResources({
+        siteId,
+        action: "read",
+        userId: ctx.user.id,
+        resourceIds: [String(resourceId)],
+      })
+
+      const { parentId } = await db
+        .selectFrom("Resource")
+        .where("id", "=", String(resourceId))
+        .where("siteId", "=", siteId)
+        .select("parentId")
+        .executeTakeFirstOrThrow()
+
+      const indexPage = await db
+        .selectFrom("Resource")
+        .where("type", "=", ResourceType.IndexPage)
+        .where("parentId", "=", parentId)
+        .select("id")
+        .executeTakeFirstOrThrow()
```

Comment:
nit: I think we can combine these two queries, seems like the first one is a subquery of the second